File: ImasLiveDB/Domain/UseCases/BackupImportSummary.swift
import Foundation

/// バックアップ復元 (引き継ぎコード / ファイル) の結果をユーザー向け文面にする。
///
/// 文面の組み立て本体は imas-core (Rust) の `domain/backup_import_summary.rs` にある。
/// 「主目的の担当/投票は 0 件でも出す」「マイタグ/スキップは入ったときだけ出す」
/// という判断と行順の安定性はそちらでテスト済み。
/// ここは Swift の `Int` を FFI の `Int64` へ橋渡しするだけの薄いラッパ
/// (生成バインディングの同名関数とは引数型で区別されるオーバーロード)。
func backupImportSummary(
    addedMarks: Int,
    addedVotes: Int,
    addedPersonalTags: Int,
    skippedMarks: Int,
    deviceIdRestored: Bool
) -> String {
    backupImportSummary(
        addedMarks: Int64(addedMarks),
        addedVotes: Int64(addedVotes),
        addedPersonalTags: Int64(addedPersonalTags),
        skippedMarks: Int64(skippedMarks),
        deviceIdRestored: deviceIdRestored
    )
}

File: imas-core/src/inbound/backup_import_summary.rs
//! バックアップ復元結果文面の FFI 面。ロジックは domain::backup_import_summary。
//!
//! 復元 1 回につき呼び出しも 1 回 (件数の集計は呼び出し側が済ませてから渡す)。

#[uniffi::export]
pub fn backup_import_summary(
    added_marks: i64,
    added_votes: i64,
    added_personal_tags: i64,
    skipped_marks: i64,
    device_id_restored: bool,
) -> String {
    crate::domain::backup_import_summary::backup_import_summary(
        added_marks,
        added_votes,
        added_personal_tags,
        skipped_marks,
        device_id_restored,
    )
}
